Revamp docs

damithc opened the pull request:
Apologies for overwriting some of the recent documentation work done by you guys. I edited freely in the interest of creating a single cohesive version of the docs. Some of the deleted content can be added back later if necessary.

Although I tried to keep the commits clean, given that there are some monster commits in the middle, I don't think it's worth reviewing these commits individually. I think the best is to go through the surge preview as if you are reading the docs for the first time and note down bugs or holes in the flow.

Side note: took about one week of continuous work. Documentation is hard work. :-p

Proposed commit message:
```
Revamp docs to leverage MarkBind features

We recently converted docs to MarkBind format. But the content was
mostly translated from the previous format which was in simple
Markdown format.

Let's revamp the docs to take advantage of MarkBind features. As this
is going to touch almost all docs, let's also take this opportunity
to improve the content where possible.

High-level overview of the changes:
* Split all-in-one-pages UG and DG into smaller pages
* Use different siteNav for UG and DG
* Add top-level pages: SHOWCASE, CONTACT
* Beautify landing page, differentiate it from the README
* Use a default layout for all pages
* Cross-embed relevant content in relevant places
* Use dark green as the brand color
* Refer to se-edu/guides for reference materials (e.g., style guides)
```

damithc commented:
@jamessspanggg how come the surge preview is missing?

jamessspanggg commented:
<img width="528" alt="Screen Shot 2020-06-06 at 12 01 02 PM" src="https://user-images.githubusercontent.com/32864116/83935575-66ae2d00-a7ed-11ea-8821-b70787403c7d.png">

It is actually because the travis build failed due to some lint errors. The deployment will only work when the travis build is successful. 

For some reason travis isn't reporting back in the status checks, happening very often for my other projects as well :/

damithc commented:
> <img alt="Screen Shot 2020-06-06 at 12 01 02 PM" width="528" src="https://user-images.githubusercontent.com/32864116/83935575-66ae2d00-a7ed-11ea-8821-b70787403c7d.png">
> 
> It is actually because the travis build failed due to some lint errors. The deployment will only work when the travis build is successful. [Link to build](https://travis-ci.org/github/reposense/RepoSense/jobs/695112843)
> 
> For some reason travis isn't reporting back in the status checks, happening very often for my other projects as well :/

Thanks @jamessspanggg will fix

jamessspanggg commented:
@damithc there's still one more error: `ERROR:docs/ug/grantingWriteAccess.mbdf:19: no newline at EOF.`

jamessspanggg commented:
woah, at first look, the docs look so much better!

damithc commented:
Build working now. Ready for review.

Tejas2805 commented:
We are following Google documentation for our guides, right?

damithc commented:
> We are following Google documentation for our guides, right?

Yes. Let me know if you spot any deviations.

damithc commented on README.md:
```diff
@@ -7,28 +7,16 @@
-* Comparison of contribution between authors
-* Filtering and sorting of authors
+* **User documentation** i.e., if you are interested in using RepoSense
+  * [**User Guide**](https://reposense.org)
```
Good point. Will add a separate link.

damithc commented:
> One thing for the overall is that I found some of the light green text content is hurting my eyes. Not sure if it is my problem or other got the same problem.

@fzdy1914 Can give an example with a screenshot? Might depend on the OS/Browser.

fzdy1914 commented:
> > One thing for the overall is that I found some of the light green text content is hurting my eyes. Not sure if it is my problem or other got the same problem.
> 
> @fzdy1914 Can give an example with a screenshot? Might depend on the OS/Browser.

![image](https://user-images.githubusercontent.com/35621726/83939324-97eb2500-a80e-11ea-80f3-693d0c8282c3.png)
This part of text with green color makes my eye not so comfortable compared with the black color one.

damithc commented:
> This part of text with green color makes my eye not so comfortable compared with the black color one.

There is a way to turn the text into black but unfortunately that turns the icon into black too. Perhaps we can get MarkBind folks to make the color apply to text only, in a future version.

![image](https://user-images.githubusercontent.com/1673303/83940240-b4d72680-a815-11ea-83cf-7539d293d0d0.png)

There is a another style (example above) that uses black color text but doesn't blend in as seamlessly as the current style.

Tejas2805 commented:
LGTM from an overview. Just 1-2 questions/suggestions. 

1) Maybe can make this consistent. 
![Screen Shot 2020-06-06 at 7 21 32 PM](https://user-images.githubusercontent.com/35946746/83943081-b14e9a00-a82b-11ea-8e59-a1886b19ef46.png)

2) The overview page of UG doesn't have a `pageNav`. If it is intended, it's okay.

Tejas2805 commented:
We might have to change some content in `HTML report` since we have added a few js files. Can do it in another PR maybe.

damithc commented on docs/dg/report.md:
```diff
@@ -0,0 +1,133 @@
+- [**v_zoom.js**](#zoom-view-v-zoom-js) - module that supports the zoom tab view
+- [**v_ramp.js**](#ramp-view-v-ramp-js) - module that supports the ramp chart view
+- [**v_segment.js**](#segment-view-v-segment-js) - module that supports the code segment view
+
```
Created #1284 for this.

damithc commented on docs/dg/workflow.md:
```diff
@@ -0,0 +1,97 @@
+**To submit a PR**, follow [this guide](https://se-education.org/guides/guidelines/PRs.html), but note the following:
+
+* As we squash the commits when merging a PR, there is ==no need to follow a strict commit organization or write elaborate commit messages for each commit==.<br>
+  However, when pushing new commits to your PR branch, do tidy up _new_ commits (i.e., commits not yet pushed) e.g., squash noise commits.
```
Good point. I removed mentions of noise commits.

damithc commented:
> * Maybe can make this consistent.
>   ![Screen Shot 2020-06-06 at 7 21 32 PM](https://user-images.githubusercontent.com/35946746/83943081-b14e9a00-a82b-11ea-8e59-a1886b19ef46.png)

Intended. I wanted to give some extra details in the first instance but didn't want to repeat the extra info in subsequent cases.

> * The overview page of UG doesn't have a `pageNav`. If it is intended, it's okay.

Intended. Doesn't add much value, plus I wanted to use full width for the main content as there are nested panels.

damithc commented:
Thanks for the reviews given so far. Fixed some. Others are either intended or to be left for separate PRs. The issue regarding the text color can be left aside for now as likely to be fixed by a future MarkBind release.

jamessspanggg commented:
Leaving 48hrs to review before merging this. @reposense/devs please take a look at this PR.

damithc commented:
Thanks @Tejas2805 I've fixed the three issues mentioned.
Also did some cosmetic tweaks to the landing page.
Found a way to simplify the landing page code in e469f26

jylee-git commented on docs/dg/settingUp.md:
```diff
@@ -0,0 +1,57 @@
+
+**Prerequisites:**
+* **JDK `1.8.0_60`** or later ([download :fas-download:](https://www.oracle.com/technetwork/java/javase/downloads/index.html)).
+* **npm** `6.0` or later ([download :fas-download:](https://www.npmjs.com/get-npm)).
```
Is NPM an explicit pre-requisite? IIRC npm is installed as a plugin when running `gradle build`

Can the other devs help me to confirm this? Thanks

damithc replied:
> With this PR, the UG link in the pic below is now dead, since the `userguide.md` is removed. Should we just link to `reposense.org`?

reposense.org/ug will be the UG for the most recent release
reposense.org/RepoSense/ug will be the UG for the latest master

jylee-git commented on docs/about.md:
```diff
@@ -1,138 +1,137 @@
----
+<!-- ------------------------------------------------------------------------------------------------------ -->
 
 ### [Anubhav](https://github.com/anubh-v)
 **Role**: Contributor [2019 September - ]<br/>
```
We can use @anubh-v 's profile pic if he wishes to do so:

```suggestion
### [Anubhav](https://github.com/anubh-v)
![](https://avatars1.githubusercontent.com/u/35621759?s=400&u=f1c240d85823576808943c93ea035166b5343a79&v=4)
**Role**: Contributor [2019 September - ]<br/>
```

anubh-v replied:
Thanks @jylee-git for your consideration and suggestion.
I would prefer to keep the current state.

anubh-v commented on docs/ug/configFiles.md:
```diff
@@ -0,0 +1,183 @@
+
+<sup>* **Multi-value column**: multiple values can be entered in this column using a semicolon `;` as the separator.</sup>
+
+> Note: the first row consists of config headings, which is ignored by RepoSense.
```
A minor comment: there is a difference in style for these two instances of the same note.

![image](https://user-images.githubusercontent.com/35621759/84052048-9c683700-a9e2-11ea-80af-7fd953c65f37.png)

We can standardise the style for the note.

anubh-v replied:
We could also remove the note entirely, as it has already been stated for `repo-config.csv`

damithc commented:
@anubh-v thanks for the review. I've moved the note about the column headings to the top of the document as it applies to all csv files.

@jylee-git We can remove the npm as a prereq if confirmed no need. In fact I added it myself because I thought it is needed but didn't have a way to confirm because I had it installed already. For the time being, we can keep it as there's no harm installing it first.

@jamessspanggg we can go ahead and merge this first. Any further improvements can be done in future PRs.

damithc commented:
Thanks again reviewers.
I've updated https://reposense.org/ with the new version of docs. Note the extra warning in DG pages. Let me know if you have any feedback on that. Probably shouldn't have capitalized 'Developer Documentation'. Will fix in the next PR.
I'll send a PR for #1285 soon.
